fsspec/implementations/tar.py

```diff
@@ -84,21 +86,39 @@ def __init__(
         self.tar = tarfile.TarFile(fileobj=self.fo)
         self.dir_cache = None
 
-        self.index_store = index_store
+        if isinstance(index_store, (str, pathlib.Path)):
```

Comment:
I think it's reasonable to only allow str here. That would allow for the index to be stored _anywhere_ (in another storage backend), whereas Path is local specific. If you use fsspec.open(), then you could include anything that it can handle.

Reply:
I like that idea. Does `fsspec` have some way to check if a file exists, akin to `pathlib.Path(...).exists()`? I could do try/except and catch for a `FileNotFoundError`, unless there is already a way to check that.

Reply:
It would be two-step, create a filesystem instance (fsspec.url_to_fs), and use .exists on that - so probably fine to use try/except here. We have discussed elsewhere whether we should have top-level operations like exists(url) that dispatch to the implementations like fsspec.open, but nothing has been done yet (cf fsspec.generic)